src/Uno.UI.Runtime.Skia.Wpf/Input/WpfKeyboardInputSource.cs

```diff
@@ -70,26 +72,43 @@ private void HostOnKeyUp(object sender, System.Windows.Input.KeyEventArgs args)
 				this.Log().Trace($"OnKeyPressEvent: {args.Key} -> {virtualKey}");
 			}
 
+			var scanCode = GetScanCode(args);
+
 			KeyUp?.Invoke(this, new(
 				"keyboard",
 				virtualKey,
 				GetKeyModifiers(args.KeyboardDevice.Modifiers),
 				new CorePhysicalKeyStatus
 				{
-					ScanCode = (uint)args.SystemKey,
+					ScanCode = scanCode,
 					RepeatCount = 1,
 				},
-				KeyCodeToUnicode((uint)args.SystemKey)));
+				KeyCodeToUnicode(scanCode)));
 		}
 		catch (Exception e)
 		{
 			Windows.UI.Xaml.Application.Current.RaiseRecoverableUnhandledException(e);
 		}
 	}
 
+	// WPF doesn't expose the scancode, but it exists internally
+	private static uint GetScanCode(KeyEventArgs args)
+	{
+		try
+		{
+			return (uint)((int)(typeof(KeyEventArgs).GetProperty("ScanCode", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)?.GetValue(args) ?? 0));
+		}
+		catch (Exception e)
+		{
+			Console.WriteLine("Couldn't get ScanCode from WPF KeyEventArgs.");
+			Console.WriteLine(e);
```

Comment:
Let's use LogError reporting. Also, let's not return a default value if the property is not available, it needs to fail explicitly.